rust-escrow/dao-factory/src/lib.rs

```diff
@@ -89,23 +83,39 @@ impl DaoFactory {
             GAS_FOR_CREATE,
         )
         .then(ext_self::on_create_callback(
-            country_code,
-            daos_by_country,
+            dao_name.clone(),
+            env::predecessor_account_id(),
+            env::attached_deposit(),
             env::current_account_id(),
             0,
             GAS_FOR_CREATE_CALLBACK,
         ))
     }
 
-    pub fn on_create_callback(&mut self, country_code: String, daos_by_country: u128) -> bool {
+    #[private]
+    pub fn on_create_callback(
+        &mut self, 
+        dao_name: String, 
+        predecessor_account_id: AccountId, 
+        attached_deposit: u128
+    ) -> bool {
         assert_eq!(env::promise_results_count(), 1, "ERR_CALLBACK_METHOD");
 
-        // handle the result from the cross contract call this method is a callback for
         match env::promise_result(0) {
-            PromiseResult::Successful(_result) => {
-                self.daos_by_country_count
-                    .insert(&(country_code), &(daos_by_country));
-                true
+            PromiseResult::Successful(result) => {
+                let res = String::from_utf8_lossy(&result);
+
+                if res == "true" {
+                    let dao_account_id: AccountId = format!("{}.{}", dao_name, env::current_account_id()).parse().unwrap();
+                    self.dao_index
+                        .insert(&predecessor_account_id, &dao_account_id);
+
+                    return true;
+                }
+
+                Promise::new(predecessor_account_id).transfer(attached_deposit);
+
+                false
```

Comment:
Same as above. What if we actually panick! instead of returning `false`?

Reply:
Yes, I think that is better instead a bool. We can catch the response easier.